### PR TITLE
[HDRP][Compositor] Documentation and bug fixes

### PR DESCRIPTION
**Pull request by pmavridis**
### Purpose of this PR
This PR includes the following:
- https://fogbugz.unity3d.com/f/cases/1283904/
Overriding the "clear depth" is not allowed in the UI for the first camera in a stack. Similarly the clear color override is only available for the first camera in the stack. The other cameras never clear the color.
- Add missing documentation regarding screen-space UI limitations in the compositor (poking @JordanL8 for visibility)
- Better defaults when creating a new camera sub-layer:  previously new sub-layers where empty (0 culling mask), so users might get confused. We now just use the culling mask of the camera and let the user override it if this is desired. 
- Small bugfixes (better handling of resizing, clone cameras when their state is overridden, etc...). 

---
### Testing status
- Compositor test is green
- The "clear depth" option is disabled for the first sub-layer in  stack
- The "clear color" option is only enabled for the first sub-layer in  stack (disabled on the rest)
- Enabling the compositor does not result in empty scene anymore (we don't override the culling musk to zero).

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

**Review by sebastienlagarde — APPROVED** [review]

File: com.unity.render-pipelines.high-definition/Documentation~/Compositor-User-Guide.md [hunk 1]
```diff
@@ -49,3 +49,7 @@ The Render Schedule is a re-orderable list of Composition Layers and Sub-layers.
 ## Composition Parameters
 This section shows every exposed property that is not an input Composition Layer (for example, a Float to control the brightness of the final composition or a Color to tint a Texture2D). In this section, the window allows you to edit each property value outside of the Composition Graph. It is good practice to expose properties from the graph to the Graphics Compositor window, instead of hard-coding their values. This helps you to share composition profiles between Projects because those you do not need to open the Composition Graph to edit any values.
 
+## Composition of screen-space UI
+The Graphics Compositor can be used to compose screen-space canvas (UI) elements. However, in this case the camera sub-layer that draws the UI:
+- should have a unique camera assigned to it. The camera should not be used in any other layer
+- should not override any camera state (antialising, culling mask, volume mask, etc). These properties can still be changed in the source camera that is used for the sub layer.
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Documentation~/Compositor-User-Guide.md:55]: - Should not override any camera state (antialiasing, culling mask, volume mask, etc). You can change these properties in the source camera that the sub-layer uses.

File: com.unity.render-pipelines.high-definition/Documentation~/Compositor-User-Guide.md [hunk 2]
```diff
@@ -49,3 +49,7 @@ The Render Schedule is a re-orderable list of Composition Layers and Sub-layers.
 ## Composition Parameters
 This section shows every exposed property that is not an input Composition Layer (for example, a Float to control the brightness of the final composition or a Color to tint a Texture2D). In this section, the window allows you to edit each property value outside of the Composition Graph. It is good practice to expose properties from the graph to the Graphics Compositor window, instead of hard-coding their values. This helps you to share composition profiles between Projects because those you do not need to open the Composition Graph to edit any values.
 
+## Composition of screen-space UI
+The Graphics Compositor can be used to compose screen-space canvas (UI) elements. However, in this case the camera sub-layer that draws the UI:
+- should have a unique camera assigned to it. The camera should not be used in any other layer
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Documentation~/Compositor-User-Guide.md:54]: - Should have a unique camera assigned to it. You should also not use the camera in any other layer.

File: com.unity.render-pipelines.high-definition/Documentation~/Compositor-User-Guide.md [hunk 3]
```diff
@@ -49,3 +49,7 @@ The Render Schedule is a re-orderable list of Composition Layers and Sub-layers.
 ## Composition Parameters
 This section shows every exposed property that is not an input Composition Layer (for example, a Float to control the brightness of the final composition or a Color to tint a Texture2D). In this section, the window allows you to edit each property value outside of the Composition Graph. It is good practice to expose properties from the graph to the Graphics Compositor window, instead of hard-coding their values. This helps you to share composition profiles between Projects because those you do not need to open the Composition Graph to edit any values.
 
+## Composition of screen-space UI
+The Graphics Compositor can be used to compose screen-space canvas (UI) elements. However, in this case the camera sub-layer that draws the UI:
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Documentation~/Compositor-User-Guide.md:53]: You can use the Graphics Compositor to compose screen-space canvas (UI) elements. However, in this case, the camera sub-layer that draws the UI:

**Review by JordanL8 — CHANGES_REQUESTED** [review]
Added a few comments for the docs.

File: com.unity.render-pipelines.high-definition/Documentation~/Compositor-User-Guide.md (hunk 2 above)
Comment (pmavridis) [reply com.unity.render-pipelines.high-definition/Documentation~/Compositor-User-Guide.md:54]: Here the second sentence is an "explanation" of what a "unique" camera means, so the term "also" is a bit confusing to me...
What about this one?:
- Should have a unique camera assigned to it. You should not use this camera in any other layer.

Comment (JordanL8) [reply com.unity.render-pipelines.high-definition/Documentation~/Compositor-User-Guide.md:54]: Oh yeah, I see that now. Your suggestion looks good!

Comment (pmavridis) [reply com.unity.render-pipelines.high-definition/Documentation~/Compositor-User-Guide.md]: Ok, should be good now.

**Review by JordanL8 — APPROVED** [review]

**Comment by sebastienlagarde** [comment]
Merging in 10.1

**Pull request merged**